### ISSUE TITLE
[Bug(ish)]: `objects_api.delete_object` does not error on non-existent object

### ISSUE THREAD

**Issue by nicholasjng** [treeverse/lakeFS#6531]
### What happened?

TLDR: `objects_api.delete_object` does not seem to throw an error on requesting a non-existent file, which I expected by its docstring.
        
Steps to Reproduce:
1. Start a lakeFS quickstart instance, populate repo with sample data.
2. Install `lakefs_client` (or `lakefs-spec`, which is used in the repro)
3. Run:

```python
    fs = LakeFSFileSystem()
    nonexistent_path = ”quickstart/main/nonexistent.md"  # <- just a wrapper around `objects_api.delete_objects()`
    fs.rm(nonexistent_path)
```

Just wanted to report this since it broke tests - this might not actually be a bug, but it is unexpected given the public info (read: docstrings).

### Expected behavior

Since this docstring says to expect a `NotFound` error:

https://github.com/treeverse/lakeFS/blob/a061cc2a121ea2e1b89ed25b8d7b130ee5d77cfe/clients/python/lakefs_client/api/objects_api.py#L1048

I expected a `NotFoundException`.

### lakeFS version

v0.108.0

### How lakeFS is installed

Docker quickstart

### Affected clients

Python lakeFS client v0.108.0

**Comment by arielshaqed** [comment]
Hi @nicholasjng ,

Perhaps unfortunately, this is actually as intended!  It is incredibly convenient to be able to receive an accurate notification of whether a `deleteObject` call deleted something or not.  Somewhat suprisingly it is _not possible_ to do so in a performant object store.  Indeed, AWS S3 also does not return whether you deleted an object or not.

The reason why we cannot do it efficiently is exactly the same as why it would be so convenient: it requires and allows incredible synchronization.  Consider 100 workers (say, a Spark cluster...) that `deleteObject` the same object concurrently.  In order to do what this issue wants, the object should no longer exist, and exactly _one_ worker should succeed.  This is a heavy-duty leader election!  Given that we _also_ want to be able to delete millions of objects, efficiency trumps usefulness here.

I'm going to close this issue, sorry: not because it's a bad idea or not useful, but because it hurts object store efficiency.

**Comment by nicholasjng** [comment]
Thanks @arielshaqed! I was aware of the design constraints and considerations, just wanted to clarify that the docstring above is inaccurate.

**Comment by arielshaqed** [comment]
> Thanks @arielshaqed! I was aware of the design constraints and considerations, ~just wanted to clarify that the docstring above is inaccurate~ (EDIT: I missed the "not").

Yeah, I was just looking and saw that.  The phrasing here is obviously confusing!  I'd love to hear a better suggestion if you can find one.
